### DISCUSSION TITLE
Response.read() takes 1 positional argument but 2 were given

### DISCUSSION THREAD

**Discussion by yaxxie-piramidal** [encode/httpx#3351] [Potential Issue]
Is the `Response`  object intended/meant to follow I/O protocols? I encountered the exception

`Response.read() takes 1 positional argument but 2 were given`

when trying to pass the response object to methods expecting I/O objects (trying to pass the read size argument)

e.g.

```python
with httpx.stream("GET", url) as r:
    resp = yaml.safe_load(r)
```

**Comment by lovelydinosaur** [comment]
That's a good pattern isn't it?
That usage isn't supported, tho it's an interesting prompt.

**Reply by yaxxie-piramidal** [reply]
> That's a good pattern isn't it?

Sorry, I didn't understand this remark.

I'm trying to understand why this usage isn't supported, given that the object **appears** to look like an object supporting i/o, and it's probably exactly what one would want when using `httpx.stream`

**Reply by lovelydinosaur** [reply]
Perhaps `.read()` should support a `size` parameter here. You're welcome to attempt a draft PR there.
